numpy/core/tests/test_multiarray.py

```diff
@@ -2256,19 +2256,23 @@ def test_searchsorted_unicode(self):
         assert_equal([a.searchsorted(a[i], 'left') for i in ind], ind)
         assert_equal([a.searchsorted(a[i], 'right') for i in ind], ind + 1)
 
-    def test_searchsorted_with_sorter(self):
+    @pytest.mark.parametrize('index, sorter, error',
+                        (
+                            (0, np.array([1, (2, 3)], dtype=object), TypeError),
+                            (0, [1.1], TypeError),
+                            (0, [1, 2, 3, 4], ValueError),
+                            (0, [1, 2, 3, 4, 5, 6], ValueError),
+                            # bounds check
+                            (4, [0, 1, 2, 3, 5], ValueError),
+                            (0, [-1, 0, 1, 2, 3], ValueError),
+                            (0, [4, 0, -1, 2, 3], ValueError),
+                        ))
+    def test_searchsorted_with_invalid_sorter(self, index, sorter, error):
         a = np.array([5, 2, 1, 3, 4])
         s = np.argsort(a)
```

Comment:
Re parameterization: seems a little wasteful to compute these each test. Not a big deal, but I'm not sure how useful this type of parameterization is.

Reply:
ok, will undo the parameterization